runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java

```diff
@@ -358,10 +360,35 @@ public void hideWidgets(WidgetHiddenChanged event)
 		{
 			return;
 		}
+		fileName = fileName + "";
 
 		takeScreenshot(fileName, config.displayDate());
 	}
 
+	@Subscribe
+	private  void onGameStateChange(GameStateChanged gameStateChanged)
+	{
+		try
+		{
+			World world = new WorldClient().lookupWorlds().findWorld(client.getWorld());
```

Comment:
At the very least, the worlds should only be loaded at plugin start. I think the xptracker plugin also does the same type of thing, maybe a client-wide world service would be a good thing to have. Regardless this is called on the same thread so you never want to do webrequests from it like this.